### ISSUE TITLE
Code after return

### ISSUE THREAD

**Issue by glsantos** [dotnet/roslyn#1189]
I know actually we have the finally keyword which works nice but we need the try block to it works.
So, my suggestion is to make it as a 'native' feature whitout needing the finally block.

Leaving the theory, here's my concept:

<pre>
<code>string MyFunction(string MyString, ref int incrementVar)
{
	return MyString.Reverse()
	{
		incrementVar++;
	};
}</code></pre>

Then, it will be interpreted (or translated) as:

<pre><code>
string MyFunction(string MyString, ref int incrementVar)
{
	var _result = MyString.Reverse();

	incrementVar++;
	
	return _result;
}</code></pre>

**Comment by svick** [comment]
This doesn't look like it would be worth it to me, your translation shows that it doesn't really save you much typing.

Also, your syntax is very confusing (and possibly hard to parse). Consider:

```c#
return new List<int>()
{
    f();
};
```

If you remove the semicolon after `f()`, then it changes from "create empty list, call `f()`, return the list" to "create list containing the result of `f()` and return it".

**Comment by gafter** [comment]
@glsantos The syntax you suggest is already legal C# and it means something different from what you want it to mean.

**Comment by dsaf** [comment]
Given you saying *"I know actually we have the finally keyword which works nice but we need the try block to it works"*.

I am surprised not to see try-finally in *"Then, it will be interpreted (or translated) as:"* sample.

**Comment by gafter** [comment]
The benefit of this proposal is too small for it to be seriously considered.
